### PR TITLE
PW-4390 Added Apache HTTP Client & added get and update paymentLink

### PR DESCRIPTION
**Pull request by saquibsayyad**
**Description**
- Added Apache HTTP client to support PATCH, GET & DELETE request methods.
- Moved create paymentLink to PaymentLinks service
- Added Get and Updated paymentLink into PaymentLinks service

**Tested scenarios**
E2E tests (including payment link create, get & update)

### CODE REVIEW

File: src/main/java/com/adyen/service/PaymentLinks.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    /**
+     * GET /paymentLinks API call
+     *
+     * @param linkId String
+     * @return PaymentLinkResource
+     * @throws IOException  IOException
+     * @throws ApiException ApiException
+     */
+    public PaymentLinkResource get(String linkId) throws ApiException, IOException {
```
Comment (Aleffio) [inline src/main/java/com/adyen/service/PaymentLinks.java:72]:
```suggestion
    public PaymentLinkResource retrieve(String linkId) throws ApiException, IOException {
```

File: src/main/java/com/adyen/util/CertificateUtil.java [hunk 2]
```diff
@@ -1,56 +0,0 @@
-/*
```
Comment (Aleffio) [inline src/main/java/com/adyen/util/CertificateUtil.java:1]: why removing this?

File: src/main/java/com/adyen/Config.java [hunk 3]
```diff
@@ -54,7 +48,7 @@
     //Terminal API Specific
     protected String terminalApiCloudEndpoint;
     protected String terminalApiLocalEndpoint;
-    protected Certificate terminalCertificate;
+    protected String terminalCertificatePath;
```
Comment (Aleffio) [inline src/main/java/com/adyen/Config.java:51]: I think this is still needed from #388

Comment (saquibsayyad) [reply src/main/java/com/adyen/Config.java:51]: ah yes, reverting it.

File: src/main/java/com/adyen/service/PaymentLinks.java (hunk 1 above)
Comment (saquibsayyad) [reply src/main/java/com/adyen/service/PaymentLinks.java]: corrected it.

File: src/main/java/com/adyen/util/CertificateUtil.java (hunk 2 above)
Comment (saquibsayyad) [reply src/main/java/com/adyen/util/CertificateUtil.java]: reverted back.

File: src/test/java/com/adyen/util/CertificateUtilTest.java [hunk 4]
```diff
@@ -1,53 +0,0 @@
-/*
```
Comment (Aleffio) [inline src/test/java/com/adyen/util/CertificateUtilTest.java:1]: could you please revert this test class as well?

Comment (saquibsayyad) [reply src/test/java/com/adyen/util/CertificateUtilTest.java]: reverted back.

**Review by Aleffio — APPROVED** [review]

**Review by andrei-dediu — APPROVED** [review]

File: src/main/java/com/adyen/httpclient/AdyenResponse.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+ * See the LICENSE file for more info.
+ */
+package com.adyen.httpclient;
+
+import java.util.List;
+import java.util.Map;
+
+class AdyenResponse {
+
+    private String response;
```
Comment (rkewlani) [inline src/main/java/com/adyen/httpclient/AdyenResponse.java:28]: May be it will be good to call this as body

Comment (saquibsayyad) [reply src/main/java/com/adyen/httpclient/AdyenResponse.java]: update it

**Review by Aleffio — APPROVED** [review]

**Review by andrei-dediu — APPROVED** [review]

**Pull request merged**